website/static/js/profile.js

```diff
@@ -563,13 +569,82 @@ var SocialViewModel = function(urls, modes) {
         return false;
     });
 
+    self.addWebsiteInput = function() {
+        this.profileWebsites.push(ko.observable().extend({
+            ensureHttp: true
+        }));
+    };
+    
+    self.removeWebsite = function(profileWebsite) {
+        var profileWebsites = ko.toJS(self.profileWebsites());
+            bootbox.confirm({
+                title: 'Remove website?',
+                message: 'Are you sure you want to remove this website from your profile?',
+                callback: function(confirmed) {
+                    if (confirmed) {
+                        var idx = profileWebsites.indexOf(profileWebsite);
+                        self.profileWebsites.splice(idx, 1);
+                        self.submit();
+                        self.changeMessage(
+                            'Website removed',
+                            'text-danger',
+                            5000
+                        );
+                        if (self.profileWebsites().length === 0) {
+                            self.addWebsiteInput();
+                        }
+                    }
+                },
+                buttons:{
+                    confirm:{
+                        label:'Remove',
+                        className:'btn-danger'
+                    }
+                }
+            });
+    };
+
     self.fetch();
 };
 SocialViewModel.prototype = Object.create(BaseViewModel.prototype);
 $.extend(SocialViewModel.prototype, SerializeMixin.prototype, TrackedMixin.prototype);
 
-var ListViewModel = function(ContentModel, urls, modes) {
+SocialViewModel.prototype.serialize = function() {
+    var serializedData = ko.toJS(this);
+    var profileWebsites = serializedData.profileWebsites;
+    serializedData.profileWebsites = profileWebsites.filter(
+        function (value) {
+            return (typeof value === 'string');
```

Comment:
As discussed, it's probably best to check truthiness here rather than type checking, since you want to filter out null, undefined, and blank strings.